src/Atmos/Model/moisture.jl

```diff
@@ -239,3 +240,131 @@ function flux_diffusive!(moist::EquilMoist, flux::Grad, state::Vars, d_q_tot)
     flux.ρu += d_q_tot .* state.ρu'
     flux.moisture.ρq_tot += d_q_tot * state.ρ
 end
+"""
+  NonEquilMoist
+
+  Assumes that there isn't any thermodynamic equilibrium
+"""
+
+struct NonEquilMoist <: MoistureModel end
+
+vars_state(::NonEquilMoist, FT) = @vars(ρq_tot::FT, ρq_liq::FT, ρq_ice::FT)
+vars_gradient(::NonEquilMoist, FT) =
+    @vars(q_tot::FT, q_liq::FT, q_ice::FT, h_tot::FT)
+vars_diffusive(::NonEquilMoist, FT) = @vars(
+    ∇q_tot::SVector{3, FT},
+    ∇q_liq::SVector{3, FT},
+    ∇q_ice::SVector{3, FT}
+)
+vars_aux(::NonEquilMoist, FT) =
+    @vars(temperature::FT, θ_v::FT, q_liq::FT, q_ice::FT, p::FT, src_qliq::FT)
+
+@inline function atmos_nodal_update_aux!(
+    moist::NonEquilMoist,
+    atmos::AtmosModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    TS = thermo_state(moist, atmos.orientation, state, aux)
+    aux.moisture.temperature = air_temperature(TS)
+    q_p_sat = PhasePartition_equil(
+        air_temperature(TS),
+        state.ρ,
+        state.moisture.ρq_tot,
+    )
+    aux.moisture.src_qliq = conv_q_vap_to_q_liq(q_p_sat, PhasePartition(TS))
+    aux.moisture.θ_v = virtual_pottemp(TS)
+    aux.moisture.q_liq = PhasePartition(TS).liq
+    aux.moisture.q_ice = PhasePartition(TS).ice
+    aux.moisture.p = air_pressure(TS)
+    return nothing
+end
+
+function thermo_state(
+    moist::NonEquilMoist,
+    orientation::Orientation,
+    state::Vars,
+    aux::Vars,
+)
+    FT = eltype(state)
+    ρinv = 1 / state.ρ
+    e_int = internal_energy(moist, orientation, state, aux)
+    q_pt = PhasePartition{FT}(
+        state.moisture.ρq_tot * ρinv,
+        state.moisture.ρq_liq * ρinv,
+        state.moisture.ρq_ice * ρinv,
+    )
+    return PhaseNonEquil{FT}(e_int, state.ρ, q_pt)
+end
+
+function gradvariables!(
+    moist::NonEquilMoist,
+    atmos::AtmosModel,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    ρinv = 1 / state.ρ
+    transform.moisture.q_tot = state.moisture.ρq_tot * ρinv
+    phase = thermo_state(moist, atmos.orientation, state, aux)
+    R_m = gas_constant_air(phase)
+    T = air_temperature(phase)
+    e_tot = state.ρe * ρinv
+    transform.moisture.h_tot = e_tot + R_m * T
```

Comment:
Computed at `AtmosModel` level, doesn't need to be carried within the moisture model  (?)